### ISSUE TITLE
ZIO 1.0.6 fails to compile implicit

### ISSUE THREAD

**Issue by juliano** [zio/zio#4992]
I have the following code running on ZIO 1.0.5:

```scala
@accessible
object Example {
  trait Service {
    def foo[A](value: String)(implicit decoder: Decoder[A]): IO[Throwable, A]
  }
}

object AnotherService {
  case class Bar(value: Int)

  object Bar {
    implicit val decoder: Decoder[Bar] = deriveDecoder
  }

  def bar(implicit decoder: Decoder[Bar]): ZIO[Has[Example.Service], Throwable, Bar] =
    Example.foo[Bar]("hey")(decoder)
}
```

After migrating to ZIO 1.0.6, it fails to compile:

```sbt
[error] /Users/juliano/dev/proj/src/main/scala/example/AnotherService.scala:14:28: not enough arguments for method foo: (implicit decoder: io.circe.Decoder[example.AnotherService.Bar], ev: izumi.reflect.Tag[example.Example.Service]): zio.ZIO[zio.Has[example.Example.Service] with Any,Throwable,example.AnotherService.Bar].
[error] Unspecified value parameter ev.
[error]     Example.foo[Bar]("hey")(Bar.decoder)
```

I've noticed it only happens when `Bar.decoder` is explicitly there, it compiles otherwise.

**Comment by adamgfraser** [comment]
@nadenf You want to take a look at this?

**Comment by nadenf** [comment]
@adamgfraser .. Looking into it now.

**Comment by adamgfraser** [comment]
@nadenf Thank you! 🙏

**Comment by nadenf** [comment]
I am actually confused how this compiles in 1.0.5 because @accessible expands:
```
def foo[A](value: String)(implicit decoder: Decoder[A]): IO[Throwable, A]
```

to:
```
def foo[A](value: String)(implicit decoder: Decoder[A])(implicit ev: implicit ev: _root_.izumi.reflect.Tag[Service])
```

And according to the Scala spec multiple implicit blocks are not allowed. @ioleo any ideas ?

Otherwise it's best just to revert #4466 and instead workaround it in the IntelliJ Plugin.

**Comment by ioleo** [comment]
It should've been combined into a single implicit parameter list, I'm suprised to see there are multiple.

**Comment by adamgfraser** [comment]
Yes I think that is right it should just be one implicit parameter list. Does someone have time to work on this?

**Comment by nadenf** [comment]
In #4466 it combines multiple implicit into a single list which is what is causing this bug.

I am fine to work on it. Just trying to get more information from the Scala compiler team.

**Comment by nadenf** [comment]
I will look to fix #4466 on the IntelliJ plugin side. 

It looks Izumi macro runs first and tries remove its implicit block which leaves the others in place.

**Comment by adamgfraser** [comment]
This has been resolved.
